### PR TITLE
Support more projects with extension

### PR DESCRIPTION
**Pull request by davilimap**
Allowing transform options to show on projects that support NuGet.

### CODE REVIEW

File: SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs [hunk 1]
```diff
@@ -47,34 +70,31 @@ public SlowCheetahNuGetManager(IServiceProvider package)
         /// <returns>True if the project supports NuGet</returns>
         public bool ProjectSupportsNuget(IVsHierarchy hierarchy)
         {
-            hierarchy.GetGuidProperty(VSConstants.VSITEMID_ROOT, (int)__VSHPROPID.VSHPROPID_ProjectIDGuid, out Guid projectGuid);
-            Project currentProject = PackageUtilities.GetAutomationFromHierarchy<Project>(hierarchy, (uint)VSConstants.VSITEMID.Root);
-            if (projectGuid == null || currentProject == null)
+            if (SupportsINugetProjectSystem(hierarchy))
```
Comment (jviau) [inline SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs]: Argument validation

File: SlowCheetah.VisualStudio/NugetHandler/VsProjectTypes.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Sayed Ibrahim Hashimi. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See  License.md file in the project root for full license information.
+
+namespace SlowCheetah.VisualStudio
+{
+    /// <summary>
+    /// VS project type GUIDs
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1600:Elements must be documented", Justification = "Static collection of GUIDs")]
+    public static class VsProjectTypes
```
Comment (jviau) [inline SlowCheetah.VisualStudio/NugetHandler/VsProjectTypes.cs]: Make this `internal`

File: SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs [hunk 3]
```diff
@@ -24,11 +24,36 @@ public class SlowCheetahNuGetManager
...
+                VsProjectTypes.CosmosProjectTypeGuid,
+                VsProjectTypes.ManagementPackProjectTypeGuid,
+            };
+
         private readonly IServiceProvider package;
 
         private readonly HashSet<string> installTasks = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
         private readonly object syncObject = new object();
 
+        private readonly Dictionary<Guid, bool> storedProjects = new Dictionary<Guid, bool>();
```
Comment (jviau) [inline SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs]: This is no longer needed.

File: SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs [hunk 4]
```diff
@@ -39,6 +64,40 @@ public SlowCheetahNuGetManager(IServiceProvider package)
         }
 
         /// <summary>
+        /// Checks if the current project supports NuGet
+        /// </summary>
+        /// <param name="hierarchy">Hierarchy of the project to be verified</param>
+        /// <returns>True if the project supports NuGet</returns>
+        public bool ProjectSupportsNuget(IVsHierarchy hierarchy)
```
Comment (jviau) [inline SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs]: argument validation.

File: SlowCheetah.VisualStudio/SlowCheetahPackage.cs [hunk 5]
```diff
@@ -586,6 +586,11 @@ private IEnumerable<string> GetPublishProfileTransforms(IVsHierarchy hierarchy,
         /// <returns>True if the project supports transformation</returns>
         private bool ProjectSupportsTransforms(IVsProject project)
         {
+            if (this.NuGetManager.ProjectSupportsNuget(project as IVsHierarchy))
```
Comment (jviau) [inline SlowCheetah.VisualStudio/SlowCheetahPackage.cs]: Can you manually verify if this works in VS 2013 and then in 2017? Also check if it works for all of the supported project extensions. If so, we can delete the old way of checking.

File: SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs [hunk 6]
```diff
@@ -39,6 +64,40 @@ public SlowCheetahNuGetManager(IServiceProvider package)
         }
 
         /// <summary>
+        /// Checks if the current project supports NuGet
+        /// </summary>
+        /// <param name="hierarchy">Hierarchy of the project to be verified</param>
+        /// <returns>True if the project supports NuGet</returns>
```
Comment (jviau) [inline SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs]: Add a `<remarks>` tag saying that this implementation is borrowed from NuGet and link to their implementation. Also mention this should be removed when/if NuGet adds this to their public API.

Comment (davilimap) [inline SlowCheetah.VisualStudio/NugetHandler/SlowCheetahNuGetManager.cs]: Done

File: SlowCheetah.VisualStudio/SlowCheetahPackage.cs (hunk 5 above)
Comment (davilimap) [inline SlowCheetah.VisualStudio/SlowCheetahPackage.cs]: It's working in VS 2013 and VS 2017

**Pull request merged**